internal/exec/stages/disks/luks.go

```diff
@@ -531,3 +564,151 @@ func (d LuksDump) hasFlag(flag string) bool {
 	}
 	return false
 }
+
+// collect the cex card domains xx.xxxx
+func getAllCexOnlineAPQNs() (string, error) {
+	apDir := "/sys/bus/ap/devices/"
+	var ret string
+	// cards details are in form of link file points to card directory.
+	event, _ := os.ReadDir(apDir)
+	if len(event) == 0 {
+		return "", fmt.Errorf("adjunct processor not found")
+	}
+	// match the regexp for pattern xx.xxxx
+	for _, dir := range event {
+		match := cexRegx.MatchString(dir.Name())
+		// if a match is found, read the uevent from that APQN directory
+		// to verify whether it corresponds to a CCA controller. This is
+		// intended for potential future use with an EP11 controller.
+		if match {
+			filePath := filepath.Join(apDir, dir.Name(), "uevent")
+			ctrl, err := getCryptoController(filePath)
+			if err != nil {
+				return "", fmt.Errorf("failed to parse uevent file: %w", err)
+			}
+			for key := range ctrl {
+				if ctrl[key] == "cca" {
+					ret += dir.Name() + ","
+				}
+			}
+			if len(ret) == 0 {
+				return "", fmt.Errorf("cannot find the cca ap controller")
+			}
```

Comment:
Shouldn't this be outside the for-loop instead?

Reply:
Done